File: apps/studentWeb/src/contracts/learningContextSubjectContract.ts
import type { SubjectContract } from './subjectContract'

export interface LearningContextSubjectContract {
  contractName: 'LearningContextSubject.v1'
  learningContextSubjectId: string
  learningContextId: string
  subjectId: string
  displayOrder: number | null
  status: 'ACTIVE' | 'INACTIVE'
  createdAt: string
  updatedAt: string
}

export interface LearningContextSubjectViewContract {
  contractName: 'LearningContextSubjectView.v1'
  association: LearningContextSubjectContract
  subject: SubjectContract
}

File: Documentacao/LIA2_008_PedagogicalAdaptive_Mobile_Thinking_FIX02/apps/backend/app/contracts/learningContextSubjectContract.py
from datetime import datetime
from typing import Literal
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field

from app.contracts.subjectContract import SubjectContract


class LearningContextSubjectCreateContract(BaseModel):
    contractName: Literal[
        "LearningContextSubjectCreate.v1"
    ] = "LearningContextSubjectCreate.v1"
    displayOrder: int | None = Field(default=None, ge=0)


class LearningContextSubjectContract(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    contractName: Literal[
        "LearningContextSubject.v1"
    ] = "LearningContextSubject.v1"
    learningContextSubjectId: UUID
    learningContextId: UUID
    subjectId: UUID
    displayOrder: int | None
    status: Literal["ACTIVE", "INACTIVE"]
    createdAt: datetime
    updatedAt: datetime


class LearningContextSubjectViewContract(BaseModel):
    contractName: Literal[
        "LearningContextSubjectView.v1"
    ] = "LearningContextSubjectView.v1"
    association: LearningContextSubjectContract
    subject: SubjectContract
